### PR TITLE
Added rotate method

### PR DESCRIPTION
**Pull request by Jibbon**
Add rotation

### CODE REVIEW

**Comment by Jibbon** [comment]
Improve rotation

**Comment by Jibbon** [comment]
Alter usage dot background colour.

**Comment by Jibbon** [comment]
Add border width option

**Comment by Jibbon** [comment]
Remove redundancies

**Comment by rossburton** [comment]
Can you rebase and squash the commits to logical improvements, so the commits are "Add rotation", "Adjust border" etc, not "do something", "fix", "revert", "change".

Easier to review, and the history tells a clean story.

Also, why the .yml?  The automatic publishing to pages works fine so far.

File: svgcard.js [hunk 1]
```diff
@@ -119,7 +121,9 @@ function renderItem(item, parent) {
       i.size(i.width() * ratio, i.height() * ratio)
        .center(width/2, ((height-35)/2)+35)
        .dmove(item.nudgeX, item.nudgeY);
-    }).css('mix-blend-mode', 'multiply');
+    }).css('mix-blend-mode', 'multiply')
+    .css('transform','rotate('+item.rotate+'deg)')
+    .css('transform-origin','center');
```
Comment (rossburton) [inline svgcard.js]: Do this using the `.transform()` method: https://svgjs.dev/docs/3.0/manipulating/#transform-as-setter

File: svgcard.js [hunk 2]
```diff
@@ -132,7 +136,14 @@ function renderItem(item, parent) {
   for (let i = 0; i < item.usage; i++) {
     const x = padding + (i % 3) * 18;
     const y = 35 + 10 + Math.floor(i / 3) * 18;
-    draw.circle(15).move(x, y).styleStroke().attr('fill', 'white').addOutline();
+    if ( item.whiteusage )
```
Comment (rossburton) [inline svgcard.js]: Code duplication: use a ternary instead:

```
.attr('fill', item.whiteusage ? 'white' : 'item.backgroundColor)
```

File: templates.js [hunk 3]
```diff
@@ -18,7 +19,9 @@ const itemDefault = {
     height: 1,
     divider: true,
     border: true,
+    bwidth: 1,
```
Comment (rossburton) [inline templates.js:22]: borderWidth, please

File: templates.js [hunk 4]
```diff
@@ -18,7 +19,9 @@ const itemDefault = {
...
     star: false,
+    whiteusage:true,
```
Comment (rossburton) [inline templates.js:24]: Vague: rename to whitePips

Comment (rossburton) [reply templates.js:24]: No, I'm just bad at reading.  Though camelcase to `whiteUsage` please.

File: svgcard.js [hunk 5]
```diff
@@ -1,5 +1,7 @@
 'use strict';
 
+var invertperc = "0";
```
Comment (rossburton) [inline svgcard.js]: Obsolete?

**Comment by Jibbon** [comment]
I've made the refinements you suggested. Thanks for showing me those tricks! But I fear I'm making a mess of things so I won't attempt any more pull requests. It's working nicely for me locally now though so thanks again!

**Comment by rossburton** [comment]
Thanks for this, it's much appreciated.

If you use `git rebase -i origin/main` you can make commits disappear or merge together, which is great so that nobody else has to see your working.

**Comment by rossburton** [comment]
I picked the changes and committed most of them to main, thanks!

I'm rewriting the HTML using bootstrap now so if you understand that then some help making it look prettier would be appreciated :)

**Pull request closed without merging**